Signal/src/ViewControllers/HomeView/ConversationSearchViewController.swift

```diff
@@ -0,0 +1,244 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+class ConversationSearchViewController: UITableViewController {
+
+    var searchResultSet: SearchResultSet = SearchResultSet.empty
+
+    var uiDatabaseConnection: YapDatabaseConnection {
+        // TODO do we want to respond to YapDBModified? Might be hard when there's lots of search results, for only marginal value
+        return OWSPrimaryStorage.shared().uiDatabaseConnection
+    }
+
+    var searcher: ConversationSearcher {
+        return ConversationSearcher.shared
+    }
+
+    enum SearchSection: Int {
+        case conversations = 0
+        case contacts = 1
+        case messages = 2
+    }
+
+    // MARK: View Lifecyle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        tableView.rowHeight = UITableViewAutomaticDimension
+        tableView.estimatedRowHeight = 60
+
+        tableView.register(ChatSearchResultCell.self, forCellReuseIdentifier: ChatSearchResultCell.reuseIdentifier)
```

Comment:
Why Chat and not `ThreadSearchResultCell` or `ConversationSearchResultCell`?

Reply:
Thread is sort of an overloaded term in any programming context. I'll change it to the longer "ConversationSearchResultCell".